### ISSUE TITLE
How to run a shell script action inside Docker container?

### ISSUE THREAD

**Issue by Vandaahl** [kopia/kopia#1884]
I'm using docker-compose to make backups with the Kopia UI. I would like to configure a 'before snapshot action' to run a shell script. It is my understanding that the Docker image does not include a shell however. Does this mean this scenario is not supported when using docker?

I was thinking about using docker-compose to combine Kopia with a [Apprise container](https://github.com/caronc/apprise) to have Kopia use a shell script with a **curl** command to post some json about successfull and failed snapshot data to the Apprise container which can pick this up and send all kinds of cool notifications.

But then again, curl isn't included in the Kopia image anyway. Maybe something to consider?

**Comment by talios** [comment]
Having native web-hook support could be useful here maybe.  In addition to specifying scripts, a set of URLs could be listed to POST events to.

**Comment by adowair** [comment]
FYI #1519 tracks a similar effort/issue

**Comment by akash07k** [comment]
@Vandaahl 
Can you please share an example of your docker-compose config?
Actually I too want to backup my docker volumes with Kopia, but no idea what how to accomplish it.
Your help will be very appreciated :-)

**Comment by akash07k** [comment]
Really thanks a lot buddy. :-)

For example, I use Linkding on my server, and I want to backup it's 
volume with Kopia, then how will I have to expose my volumes to the 
underlying Kopia image?

My example docker-compose.yml is below (for now I'm using 
docker-volume-backup image but I want to use Kopia instead of it):

```

version: '3'

services:
   app:
     image: sissbruecker/linkding:latest
     ports:
       - "${LD_HOST_PORT:-9090}:9090"

     labels:
       - docker-volume-backup.stop-during-backup=true

     volumes:
       - /etc/timezone:/etc/timezone:ro
       - /etc/localtime:/etc/localtime:ro
       - "${LD_HOST_DATA_DIR:-./data}:/etc/linkding/data"
     restart: unless-stopped

   backup:
     image: offen/docker-volume-backup:latest
     restart: unless-stopped
     env_file: ./backup.env

     environment:
       - AWS_S3_PATH=${COMPOSE_PROJECT_NAME}

     volumes:
       - /etc/timezone:/etc/timezone:ro
       - /etc/localtime:/etc/localtime:ro
       - 
./notifications.tmpl:/etc/dockervolumebackup/notifications.d/notifications.tmpl
       - ${LD_HOST_DATA_DIR:/backup/linkding:ro
       - /var/run/docker.sock:/var/run/docker.sock:ro
       - /${HOME}/backups/${COMPOSE_PROJECT_NAME}:/archive
```

**Comment by jkowalski** [comment]
Fixed in #2252
